lib/internal/Magento/Framework/ObjectManager/Factory/AbstractFactory.php

```diff
@@ -92,114 +92,119 @@ public function setArguments($arguments)
-                );
-            case 10:
-                return new $type(
-                    $args[0], $args[1], $args[2], $args[3], $args[4], $args[5], $args[6], $args[7], $args[8], $args[9]
-                );
-            case 11:
-                return new $type(
-                    $args[0],
-                    $args[1],
-                    $args[2],
-                    $args[3],
-                    $args[4],
-                    $args[5],
-                    $args[6],
-                    $args[7],
-                    $args[8],
-                    $args[9],
-                    $args[10]
-                );
-            case 12:
-                return new $type(
-                    $args[0],
-                    $args[1],
-                    $args[2],
-                    $args[3],
-                    $args[4],
-                    $args[5],
-                    $args[6],
-                    $args[7],
-                    $args[8],
-                    $args[9],
-                    $args[10],
-                    $args[11]
-                );
-            case 13:
-                return new $type(
-                    $args[0],
-                    $args[1],
-                    $args[2],
-                    $args[3],
-                    $args[4],
-                    $args[5],
-                    $args[6],
-                    $args[7],
-                    $args[8],
-                    $args[9],
-                    $args[10],
-                    $args[11],
-                    $args[12]
-                );
-            case 14:
-                return new $type(
-                    $args[0],
-                    $args[1],
-                    $args[2],
-                    $args[3],
-                    $args[4],
-                    $args[5],
-                    $args[6],
-                    $args[7],
-                    $args[8],
-                    $args[9],
-                    $args[10],
-                    $args[11],
-                    $args[12],
-                    $args[13]
-                );
-            case 15:
-                return new $type(
-                    $args[0],
-                    $args[1],
-                    $args[2],
-                    $args[3],
-                    $args[4],
-                    $args[5],
-                    $args[6],
-                    $args[7],
-                    $args[8],
-                    $args[9],
-                    $args[10],
-                    $args[11],
-                    $args[12],
-                    $args[13],
-                    $args[14]
-                );
-            default:
-                $reflection = new \ReflectionClass($type);
-                return $reflection->newInstanceArgs($args);
+//      TODO:remove if after end support 5.5 version of php  
+        if (version_compare(phpversion(), '5.6.0', '>='))) {
+            return new $type(... $args); // will work in PHP >= 5.6
```

Comment:
This is invalid syntax for PHP < 5.6 so this will lead to fatal error. When minimal supported version of Magento became 5.6 this PR may be merged if last commit will be rolled back.

So since minimal supported PHP version became greater than 5.6 method body should contains only
`return new $type(... $args);`
Until that time there is no way to avoid reflection (any workaround will not have performance benefits)